python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -767,6 +767,30 @@ cdef class Column:
             c_result = make_unique[column](self.view())
         return Column.from_libcudf(move(c_result))
 
+    cpdef size_type device_buffer_size(self):
+        """
+        The total size of the device buffers used by the Column.
+
+        Notes
+        -----
+        Since Columns relies on Python memoryview-like semantics to maintain
+        shared ownership of the data, the device buffers underlying this column
+        might be shared between other data structures including other columns.
+
+        Returns
+        -------
+        Number of bytes.
+        """
+        cdef size_type ret = 0
+        if self.data() is not None:
+            ret += self.data().nbytes
+        if self.null_mask() is not None:
+            ret += self.null_mask().nbytes
+        if self.children() is not None:
+            for child in self.children():
+                ret += (<Column?>child).device_buffer_size()
+        return ret
```

Comment:
```suggestion
        cdef uint64_t ret = 0
        if self.data() is not None:
            ret += self.data().nbytes
        if self.null_mask() is not None:
            ret += self.null_mask().nbytes
        if self.children() is not None:
            for child in self.children():
                ret += (<Column?>child).device_buffer_size()
        return ret
```

Reply:
Note, annoyingly. The `nbytes` property of `gpumemoryview` objects is _not_ typed (and potentially expensive since it inspects the cuda array interface dict and calls a functools method).

Can you do something like this in gpumemoryview (we need the `nbytes` property for dask comms interop), obviously the `pxd` needs to have the definitions.

```
cdef class gpumemoryview:
    cdef uint64_t _buffer_size
    cpdef uint64_t device_buffer_size(self):
          return self._buffer_size
    @property
    def nbytes(self):
        return self._buffer_size
```
And move the initialisation of `_buffer_size` to the `__init__` method?

Reply:
Good catch, done in https://github.com/rapidsai/cudf/pull/18602/commits/018a9de8256126d4d09a766d45df2d8ca6ae1d2c